cardano-api/src/Cardano/Api/Script.hs

```diff
@@ -1303,42 +1151,38 @@ instance ToJSON (SimpleScript lang) where
            ]
 
 
-instance IsSimpleScriptLanguage lang => FromJSON (SimpleScript lang) where
-  parseJSON = parseSimpleScript simpleScriptVersion
-
+instance FromJSON SimpleScript where
+  parseJSON = parseSimpleScript
 
-parseSimpleScript :: SimpleScriptVersion lang
-                  -> Value -> Aeson.Parser (SimpleScript lang)
-parseSimpleScript lang v = parseScriptSig          v
-                       <|> parseScriptBefore  lang v
-                       <|> parseScriptAfter   lang v
-                       <|> parseScriptAny     lang v
-                       <|> parseScriptAll     lang v
-                       <|> parseScriptAtLeast lang v
+-- TODO: Left off here. You need to comb through cardano-api's property tests concerning simple scripts
```

Comment:
The TODO could use some more elaboration for the benefit of the developer who will end up implementing it.